src/daemon/mod.rs

```diff
@@ -411,6 +411,20 @@ pub(super) async fn start(
             )
             .await
         });
+
+        let finality = chain_config.policy.chain_finality;
+        let chain = config.chain.to_string();
+        services.spawn_blocking(move || {
+            crate::f3::run_f3_sidecar_if_enabled(
+                format!("http://{rpc_address}/rpc/v1"),
+                crate::rpc::f3::get_f3_rpc_endpoint().to_string(),
```

Comment:
In general, it would be nice if `get_f3_rpc_endpoint` would return a `Url` and the strong typing would be used in other places.

Reply:
Changed to returning `SocketAddr`

Reply:
That's a weird one. We could make an enum, but if it's too much hassle and overengineering we could do with a type alias `F3Endpoint` for the time being. It's not blocking.